exporter/cortexexporter/helper.go

```diff
@@ -0,0 +1,213 @@
+		l[lb.Key] = prompb.Label{
+			Name:  sanitize(lb.Key),
+			Value: lb.Value,
+		}
+	}
+
+	for i := 0; i < len(extras); i += 2 {
+		if i+1 >= len(extras) {
+			break
+		}
+		_, found := l[extras[i]]
+		if found {
+			log.Println("label " + extras[i] + " is overwritten. Check if Prometheus reserved labels are used.")
+		}
+		l[extras[i]] = prompb.Label{
+			Name:  sanitize(extras[i]),
+			Value: extras[i+1],
+		}
+	}
+
+	s := make([]prompb.Label, 0, len(l))
+
+	for _, lb := range l {
+		s = append(s, lb)
+	}
+
+	return s
+}
+
+// getPromMetricName creates a Prometheus metric name by attaching namespace prefix, and _total suffix for Monotonic
+// metrics.
+func getPromMetricName(desc *otlp.MetricDescriptor, ns string) string {
+
+	if desc == nil {
+		return ""
+	}
+	// whether _total suffix should be applied
+	isCounter := desc.Type == otlp.MetricDescriptor_MONOTONIC_INT64 ||
+		desc.Type == otlp.MetricDescriptor_MONOTONIC_DOUBLE
+
+	b := strings.Builder{}
+
+	fmt.Fprintf(&b, ns)
+
+	if b.Len() > 0 {
+		fmt.Fprintf(&b, delimeter)
+	}
+	fmt.Fprintf(&b, sanitize(desc.GetName()))
+
+	// Including units makes two metrics with the same name and label set belong to two different TimeSeries if the
+	// units are different.
+	/*
+		if b.Len() > 0 && len(desc.GetUnit()) > 0{
+			fmt.Fprintf(&b, delimeter)
+			fmt.Fprintf(&b, desc.GetUnit())
+		}
+	*/
+
+	if b.Len() > 0 && isCounter {
+		fmt.Fprintf(&b, delimeter)
+		fmt.Fprintf(&b, totalStr)
+	}
+	return b.String()
+}
+
+
+// Simple helper function that takes the <Signature String - *TimeSeries> map
+// and creates a WriteRequest from the struct -- can move to the helper.go file
+func wrapTimeSeries(TsMap map[string]*prompb.TimeSeries) (*prompb.WriteRequest, error) {
+	if len(TsMap) == 0 {
+		return nil, fmt.Errorf("invalid TsMap: cannot be empty map")
+	}
+	TsArray := []prompb.TimeSeries{}
+	for _, v := range TsMap {
+		TsArray = append(TsArray, *v)
+	}
+	wrapped := prompb.WriteRequest{
+		Timeseries: TsArray,
+		//Other parameters of the WriteRequest are unnecessary for our Export (assumption --will revise if needed)
+	}
+	return &wrapped, nil
+}
+
+// copied from prometheus-go-metric-exporter
+// sanitize replaces non-alphanumeric characters with underscores in s.
+func sanitize(s string) string {
+	if len(s) == 0 {
+		return s
+	}
+
+	// Note: No length limit for label keys because Prometheus doesn't
+	// define a length limit, thus we should NOT be truncating label keys.
+	// See https://github.com/orijtech/prometheus-go-metrics-exporter/issues/4.
+
+	s = strings.Map(sanitizeRune, s)
+	if unicode.IsDigit(rune(s[0])) {
+		s = keyStr + delimeter + s
+	}
+	if s[0] == '_' {
+		s = keyStr + s
+	}
+	return s
+}
+
+// copied from prometheus-go-metric-exporter
+// sanitizeRune converts anything that is not a letter or digit to an underscore
+func sanitizeRune(r rune) rune {
+	if unicode.IsLetter(r) || unicode.IsDigit(r) {
+		return r
+	}
+	// Everything else turns into an underscore
+	return '_'
+}
```

Comment:
Add a blank line here, it is a code standard to always leave a final blank line.